packages/server/src/api.ts

```diff
@@ -26,6 +28,39 @@ import { io } from "./socket_io";
 export const router = express.Router();
 
 // Set up express routes
+
+router.get('/game/:gameId/sgf', async (req, res) => {
+  try {
+
+    const game = await getGame(req.params.gameId);
+
+    const game_obj = makeGameObject(game.variant, game.config);
+
+    game.moves.forEach((moves) => {
+      const { player, move } = getOnlyMove(moves);
+      game_obj.playMove(player, move);
+    });
+
+    if (game_obj.getSGF() === '') {
+      throw new Error(`SGF not supported for variant ${game.variant}`);
+    } else if (game_obj.getSGF() === "non-rectangular"){
+      throw new Error('SGF for non rectangular boards is not possible');
+    } else if (!(game_obj.phase === 'gameover')){
```

Comment:
For my own understanding is there something about SGF that requires the game to be finished? This is totally fine regardless, just wondering if we should consider implementing mid-game sgf download in the future.

Reply:
We could definitely implement a mid-game sgf download if needed; it wouldn't require that much modification.

Reply:
> We could definitely implement a mid-game sgf download if needed; it wouldn't require that much modification.



> We could definitely implement a mid-game sgf download if needed; it wouldn't require that much modification.

Now that I think about that, our current implementation relies on the game to be over before it saves the entire sgf. It is still possible but would require a different approach